vnet/lib/vnet/openflow/ovs_ofctl.rb

```diff
@@ -17,17 +17,30 @@ def initialize(datapath_id)
       @ovs_ofctl = 'ovs-ofctl -O OpenFlow13'
       @ovs_ofctl_10 = 'ovs-ofctl -O OpenFlow10'
       @ovs_vsctl = 'ovs-vsctl'
-      @ovs_vsctl += " --db=#{conf.ovsdb}" if conf.ovsdb
+
+      if conf.ovsdb
+        `ovsdb-client list-dbs #{conf.ovsdb}`
+        raise "Unable to connect to OVSDB at #{conf.ovsdb}." if $?.exitstatus != 0
+
+        @ovs_vsctl += " --db=#{conf.ovsdb}"
+      end
 
       @switch_name = conf.switch || get_bridge_name(datapath_id)
 
+      `#{@ovs_ofctl} show #{@switch_name}`
+      raise "Unable to connect to switch #{@switch_name}" if $?.exitstatus != 0
```

Comment:
With a raise with no proper catch, if you do a quick enough if up / down on a bridge interface it would crash vna. This is an issue when vna's run handles multiple datapaths, or when a crash is something humans need to check.

Though not really a priority issue to fix imo.